fw/cache.c

```diff
@@ -2737,12 +2744,38 @@ tfw_cache_build_resp(TfwHttpReq *req, TfwCacheEntry *ce, long lifetime,
 	if (tfw_cache_set_status(db, ce, resp, &trec, &p, &h_len))
 		goto free;
 
+	pskb_head = resp->msg.skb_head;
+	pmit = nmit = resp->mit;
+	nmit.start_off = 0;
+	nmit.iter.skb = nmit.iter.skb_head = NULL;
+	nmit.iter.frag = -1;
+
 	for (h = TFW_HTTP_HDR_REGULAR; h < ce->hdr_num; ++h) {
 		bool skip = !TFW_MSG_H2(req) && (h >= ce->hdr_h2_off);
+		bool is_trailer_hdr = ((TfwCStr *)p)->flags & TFW_CSTR_TRAILER;
+		int r;
+
+		if (is_trailer_hdr && TFW_MSG_H2(req)) {
```

Comment:
Looks hacky for me. I would prefer to change cache entry layout this way:
```
 Cache entry {
 [HTTP1/HTTP2 headers]
 [HTT2 only HEADERS]
 [DATA]
 [Trailer headers]
 } 
```

Count of ` [HTTP1/HTTP2 headers]` +  `[HTT2 only HEADERS]` stored in `TfwCacheEntry::hdr_num`, for `[Trailer headers]` we can introduce new filed in `TfwCacheEntry`. To do this we can break `TfwCacheEntry::hdr_num` into two unsigned shorts. `unsigned short TfwCacheEntry::hdr_num` and `unsigned short `TfwCacheEntry::trailer_hdr_num`. Before building the body we iterate over headers until `TfwCacheEntry::hdr_num`. After building the body you need just iterate over the `TfwCacheEntry::trailer_hdr_num`.

While writing data to cache, we can try to adopt `TfwHttpTransIter::map`, for working
with trailers. For instance just add offset where trailer headers starts(Seems this should work, because trailers always at the end).

Reply:
@EvgeniiMekhanik @const-t @krizhanovsky 
Yes, of course, I know what you're talking about, we can do it in a normal way, which was also my option.

But as I said before, in my assessment, this requires more work as it is more complex than the hacking way and is also error-prone as it involves more parts of the framework and I need to modify and add more code, such as the current parser hardcore hdr table.

Why did I choose the hacking way? Because it's KISS. I wonder if it's worth doing a massive refactor just for the trailer issue. Anyway, I won't reject any options, I will only code for your final decision.

So anyway, I think we need a decision here, rather than the tech detail.